cisco_aci/datadog_checks/cisco_aci/tags.py

```diff
@@ -9,66 +9,112 @@
 from . import helpers
 from . import exceptions
 
+TN_REGEX = re.compile('/tn-([^/]+)/')
+APP_REGEX = re.compile('/ap-([^/]+)/')
+
 
 class CiscoTags:
-    def __init__(self, check):
-        self.check = check
+    def __init__(self, log=None):
         self.tenant_farbic_mapper = {}
         self.tenant_tags = {}
         self._api = None
+        if log:
+            self.log = log
+        else:
+            import logging
+            self.log = logging.getLogger('cisco_api')
 
-    def app_tags(self, app):
+    def _app_tags(self, app):
         tags = []
+        if not app or type(app) is not dict:
+            return tags
         attrs = app.get('attributes', {})
+        if type(attrs) is not dict:
+            return tags
         app_name = attrs.get('name')
         dn = attrs.get('dn')
         if app_name:
             tags.append("application:" + app_name)
         if dn:
-            tenant = re.search('/tn-([a-zA-Z-_0-9]+)/', dn)
+            tenant = re.search(TN_REGEX, dn)
             if tenant:
                 tags.append("tenant:" + tenant.group(1))
         return tags
 
-    def tenant_mapper(self, edpt):
-        tags = []
+    def _edpt_tags_map(self, edpt):
+        tags_map = {}
+        if not edpt or type(edpt) is not dict:
+            return tags_map
         attrs = edpt.get('attributes', {})
-        epg_name = attrs.get('name')
+        if type(attrs) is not dict:
+            return tags_map
+        epg_name_tmp = attrs.get('name')
         dn = attrs.get('dn')
-        application_meta = [
-            "endpoint_group:" + epg_name
-        ]
+        if epg_name_tmp:
+            epg_name = epg_name_tmp
+            tags_map["endpoint_group"] = "" + epg_name  # type enforcement
```

Comment:
you don't need to worry about this (unless you found a bug or something)

Reply:
Going to leave it as is because it makes the api more stable. However, I will remove the useless tmp var.